Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -465,11 +528,18 @@ struct VersionAssignmentSet<C: PackageContainer>: Equatable, Sequence {
             // A package can be excluded if there are no constraints on the
             // package (it has not been requested by any other package in the
             // assignment).
-            return constraints[container.identifier] == .any
+            return constraints[container.identifier] == .versionSet(.any)
 
         case .version(let version):
             // A version is valid if it is contained in the constraints.
-            return constraints[container.identifier].contains(version)
+            if case .versionSet(let versionSet) = constraints[container.identifier] {
+                return versionSet.contains(version)
+            }
+            return false
```

Comment:
Actually this is correct (I think?) This basically says this:

```swift
switch constraints[container.identifier] {
    case .versionSet(let versionSet):
        return versionSet.contains(version)
    case .unversioned: 
        return false
}
```

i.e. A binding versioned assignment is not valid if current assignment is unversioned